### ISSUE TITLE
nginx web directory not working

### ISSUE THREAD

**Issue by Vivek230800** [xbmc/xbmc#22204]
### Describe the bug
Nginx http or https web directory server not working

Here is a clear and concise description of what the problem is:

When I add a network location (local or global domain) it accepts it but all the files on the home page lists as 0kb. And there is nothing in the folders and I cannot see any subfolders or files within them. 

I have tried accessing it through linux terminal and with it I can see it but only not in kodi.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
Kodi is supposed to scan the files so that I see it in my movies, tv shows and music library.

Kodi is only taking 4 seconds to scan the network location on a 400gb fileserver web directory. 

## Actual Behavior

Kodi adds the directory however I can only see the folders and files on the homepage but only with their names. There is nothing in  any of the folders and the files are 0kb.

### To Reproduce

1. I tried adding the network location with http and https web directory but it is not working by any ports.
2. Then I directly added the network location from the browse location menu. And it is having the problem I have described above.
3.

### Debuglog
 you can find the complete log file here. please search the specific logs by parent directory "https://vivteamdrive.uboti.workers.dev/"

https://gist.github.com/Vivek230800/4f148e6c623396467b64f3b2a6bd2108

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Please check the parent path in the debug log you see above. You can try visiting my web directory here from the browser"https://vivteamdrive.uboti.workers.dev/"  . I want to add that web directory on kodi.

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [x] Windows
 - [x] Windows UWP

 - Operating system version/name: windows11/matrix 19.4 & android12/matrix 19.4
 - Kodi version: Matrix 19.4 used in both the oses

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by Vivek230800** [comment]
Or try adding this web directory as well. https://goindex-classic.teamsdrives.workers.dev/

**Comment by arnova** [comment]
Try disabling fancy-indexing.

**Comment by arnova** [comment]
It looks like the file listings are not generated as html but using javascript, I think that's what's causing the problem....

**Comment by enen92** [comment]
Yeah the last provided link is to ignore, directory is generated by javascript. If you want that as a source in kodi you need to create an addon to parse it.

The first link sounds like a bug in Kodi related to base paths. The following patch seems to fix it for me:

```
diff --git a/xbmc/filesystem/HTTPDirectory.cpp b/xbmc/filesystem/HTTPDirectory.cpp
index 0097b7f15c..3fdca1f35e 100644
--- a/xbmc/filesystem/HTTPDirectory.cpp
+++ b/xbmc/filesystem/HTTPDirectory.cpp
@@ -170,21 +170,28 @@ bool CHTTPDirectory::GetDirectory(const CURL& url, CFileItemList &items)
        * the CURL::Decode above
        * - https://tools.ietf.org/html/rfc1808#section-5.3
        */
-      auto NameMatchesLink([](const std::string& name, const std::string& link) -> bool
-      {
-        return (name == link) ||
-               ((std::string::npos != name.find(':')) && (std::string{"./"}.append(name) == link));
-      });
+      auto NameMatchesLink(
+          [](const std::string& name, const std::string& basePath, const std::string& link) -> bool
+          {
+            return (name == link) || CURL::Decode(std::string(basePath).append(name)) == link ||
+                   ((std::string::npos != name.find(':')) &&
+                    (std::string{"./"}.append(name) == link));
+          });
 
       // we detect http directory items by its display name and its stripped link
       // if same, we consider it as a valid item.
-      if (strLinkTemp != ".." && strLinkTemp != "" && NameMatchesLink(strNameTemp, strLinkTemp))
+      if (strLinkTemp != ".." && strLinkTemp != "" &&
+          NameMatchesLink(strNameTemp, strBasePath, strLinkTemp))
       {
         CFileItemPtr pItem(new CFileItem(strNameTemp));
         pItem->SetProperty("IsHTTPDirectory", true);
         CURL url2(url);
 
-        url2.SetFileName(strBasePath + strLinkBase);
+        if (!StringUtils::StartsWith(strLinkBase, strBasePath))
+          url2.SetFileName(strBasePath + strLinkBase);
+        else
+          url2.SetFileName(strLinkBase);
+
         url2.SetOptions(strLinkOptions);
         pItem->SetURL(url2);
 ```

Not sure I want to drive this forward though..
